### PR TITLE
Feature/test batch size 1 issue

### LINKED ISSUE #1013: Bug: `Metrics` class fails to recognize the number of samples the training dataset contains when it is of size 1
**Bug description:**
When using the testing facility, `fedbiomed.common.metrics.Metrics` fails to recognize the number of samples of `y_pred` and `y_true` have, in the method `_configure_y_true_pred`. This raise the error  `Predictions and true values should have equal number of samples`, because the check `len(y_pred) == len(y_true)` fails, when using MNIST dataset. Indeed, `y_pred` is of size `(1, 10)`  - probability of each of the 10 digits - whereas `y_true` of size `(1,1)` so the check gives `len(y_pred)=10` and `len(y_true)=1`. Check fails albeit `y_true` and `y_pred` have correct sizes.

**Suggestions**
we can pass the number of samples as argument, or handle specifically `y_pred` and `y_true` when the testing dataset is equal to `1` or improve the batch_size inference

### PR DESCRIPTION
**Pull request by srcansiz**
**PR description**

Closes  #1013 

**Developer Certificate Of Origin (DCO)**

By opening this merge request, you agree to the
[Developer Certificate of Origin (DCO)](https://github.com/fedbiomed/fedbiomed/blob/master/CONTRIBUTING.md#fed-biomed-developer-certificate-of-origin-dco)

This DCO essentially means that:

- you offer the changes under the same license agreement as the project, and
- you have the right to do that,
- you did not steal somebody else’s work.

**License**

Project code files should begin with these comment lines to help trace their origin:
```
# This file is originally part of Fed-BioMed
# SPDX-License-Identifier: Apache-2.0
```

Code files can be reused from another project with a compatible non-contaminating license.
They shall retain the original license and copyright mentions.
The `CREDIT.md` file and `credit/` directory shall be completed and updated accordingly.

**Guidelines for PR review**

General:

* give a glance to [DoD](https://fedbiomed.org/latest/developer/definition-of-done/)
* check [coding rules and coding style](https://fedbiomed.org/latest/developer/usage_and_tools/#coding-style)
* check docstrings (eg run `tests/docstrings/check_docstrings`)

Specific to some cases:

* update all conda envs consistently (`development` and `vpn`, Linux and MacOS)
* if modified researcher (eg new attributes in classes) check if breakpoint needs update (`breakpoint`/`load_breakpoint` in `Experiment()`, `save_state_breakpoint`/`load_state_breakpoint` in aggregators, strategies, secagg, etc.)
* if modified a component with versioning (config files, breakpoint, messaging protocol) then update the version following the rules in `common/utils/_versions.py`

### CODE REVIEW

File: notebooks/sklearn-perceptron.ipynb [hunk 1]
```diff
@@ -244,7 +244,7 @@
    "outputs": [],
    "source": [
     "n3 = np.concatenate((C3, y3), axis=1)\n",
-    "np.savetxt('./data/perp-c3.csv',n3,delimiter=',')"
+    "np.savetxt('../data/perp-c3.csv',n3,delimiter=',')"
```
Comment (ybouilla) [inline notebooks/sklearn-perceptron.ipynb:247]: Was there a good reason to change the path of the datasets?

File: notebooks/101_getting-started.ipynb [hunk 2]
```diff
@@ -111,13 +111,20 @@
...
     "    'loader_args': { 'batch_size': 48, }, \n",
     "    'optimizer_args': {\n",
     "        \"lr\" : 1e-3\n",
     "    },\n",
+    "    'test_ratio' : 0.25,\n",
+    "    'test_batch_size': 64,\n",
+    "    'test_metric': MetricTypes.F1_SCORE,\n",
+    "    'test_on_global_updates': True,\n",
+    "    'test_on_local_updates': True,\n",
+    "    'test_metric_args': {'average': 'marco'},\n",
```
Comment (ybouilla) [inline notebooks/101_getting-started.ipynb:127]: I feel like we have to keep this notebook as simple as possible. I dont think adding parameters related to validation is a good idea here

**Comment by ybouilla** [comment]
I think it is missing one or several tests with `test_batch_size=1` just to check issue has been solved

File: notebooks/sklearn-perceptron.ipynb (hunk 1 above)
Comment (srcansiz) [reply notebooks/sklearn-perceptron.ipynb:247]: Hi @ybouilla,
It is just that I keep data directory in root fedbiomed, and I forgot to revert it back before creating this PR.

**Comment by srcansiz** [comment]
Hi @ybouilla,

Thank you for the review. I applied your suggestion for the PR. I added `batch_size=1` cases in unit tests to make sure that it works for different use cases.

**Review by ybouilla — COMMENTED** [review]
Thanks for this PR:
I tested all the notebooks in the tutorial section that could be tested (ie for csv and 2D images) : and all of them work! Congrats for this PR.
Let me know if the changes I proposed you in this new commit suits you

File: tests/test_metrics.py [hunk 3]
```diff
@@ -45,84 +64,135 @@ def test_metrics_02_evaluate_binary_classification_1D_array(self):
...
+            ),
+            'batch_size_1': (
+                [[0]],
+                [[0.8, .1, .1]]
+            ),
+
+        }
+        for y_true, y_pred in examples.values():
+            self._check_metrics_resuls(y_true, y_pred, self._classification_metrics, 1, "binary")
+
```
Comment (ybouilla) [inline tests/test_metrics.py:193]: sorry for this error: should be "mutliclass" here

**Pull request merged**